toolkit/tools/internal/network/network.go

```diff
@@ -117,11 +133,49 @@ func DownloadFileWithRetry(ctx context.Context, srcUrl, dstFile string, caCerts
 }
 
 // DownloadFile downloads `url` into `dst`. `caCerts` may be nil. If there is an error `dst` will be removed.
-func DownloadFile(ctx context.Context, url, dst string, caCerts *x509.CertPool, tlsCerts []tls.Certificate) (err error) {
+// First attempts HTTP download, then falls back to Azure SDK if the URL is an Azure Blob Storage URL.
+func DownloadFile(
+	ctx context.Context,
+	url, dst string,
+	caCerts *x509.CertPool,
+	tlsCerts []tls.Certificate,
+	azureClientID string,
```

Comment:
Please don't bundle up blob store access with a simple HTTP download into this function. This function is meant to handle one simple case and if we need to include Azure blob store integration, then we should do this on a higher level code.

Reply:
Decision to not implement fallback but to add a second download method instead